src/NodeHelper.groovy

```diff
@@ -0,0 +1,1136 @@
+            DumbSlave newSlave = new DumbSlave(
+                        newNodeName,
+                        newNodeRemoteFS,
+                        launcher
+                        );
+
+            newSlave.setNumExecutors(newNodeNumExecutors);
+            newSlave.setNodeDescription(newNodeDescription);
+            ((Slave)newSlave).setMode(newNodeMode);
+            newSlave.setLabelString(newNodeLabelString);
+
+            (Jenkins.getInstance()).addNode(newSlave);
+
+            (Jenkins.getInstance().getComputer(newNodeName)).connect(false);
+
+            ret = newNodeName;
+        }
+
+        return ret;
+    } 
+
+    /**
+     * Overwrites the existing labels with the ones passed
+     * in. Labels should be seperated by spaces.
+     *
+     * @param computerName the computer whose labels need to be updated
+     * @param label        the net label(s) for the computer
+     *
+     * @return string of labels set on the computer
+     */
+    public String addLabel(String computerName, String label) {
+        String ret = "addLabel:COMPUTER_NOT_FOUND";
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            computer.getNode().setLabelString(label.toLowerCase());
+            ret = getLabels(computer.getName());
+        }
+
+        return ret;
+    }
+
+    /**
+     * Appends label(s) to the set of existing labels. Labels
+     * should be seperated by spaces
+     *
+     * @param computerName the computer whose labels need to be updated
+     * @param label        the net label(s) for the computer
+     *
+     * @return string of labels set on the computer
+     */
+    public String appendLabel(String computerName, String label) {
+        String ret = "appendLabel:COMPUTER_NOT_FOUND";
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            ret = addLabel(computer.getName(), getLabels(computer.getName()) + " " + label);
+        }
+
+        return ret;
+    }
+
+    /**
+     * A helper function that gets the labels from the
+     * computer
+     *
+     * @param compterName computer for which labels are needed
+     *
+     * @return labels as string seperated by space
+     */
+    public String getLabels(String computerName) {
+        String ret = "getLabels:COMPUTER_NOT_FOUND";
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            ret = (computer.getNode()).getLabelString();
+        }
+
+        return ret;
+    }
+
+    /**
+     * Creates jenkins labels. Labels are seperated by
+     * spaces. At the moment, it will add os, architecture
+     * and kernel labels. Endian lable will only be added
+     * if running on ppc.
+     *
+     * @param compterName computer for which labels are needed
+     *
+     * @return label string
+     */
+    public String constructLabels(String computerName) {
+        /* Procedure for adding support for new labels
```

Comment:
Haul comment up

Reply:
I have left that comment inside because it's for anyone who wants to add more functionality to the function. It isn't necessarily needed in the header comment of the function